### PR TITLE
feat: Convinience method for AccessToken expiration

### PR DESCRIPTION
**Pull request by davhdavh**

### CODE REVIEW

File: src/OpenIddict.Client/OpenIddictClientModels.cs [hunk 1]
```diff
@@ -714,6 +720,11 @@ public sealed record class RefreshTokenAuthenticationResult
         /// </summary>
         public required string AccessToken { get; init; }
 
+        /// <summary>
+        /// Gets or sets the access token expiration date, if available.
```
Comment (kevinchalet) [inline src/OpenIddict.Client/OpenIddictClientModels.cs]:
```suggestion
        /// Gets or sets the expiration date of the access token, if available.
```

(for consistency with the comment used for `ProcessAuthenticationContext.BackchannelAccessTokenExpirationDate`)

File: src/OpenIddict.Client/OpenIddictClientModels.cs [hunk 2]
```diff
@@ -314,6 +319,7 @@ public sealed record class ClientCredentialsAuthenticationResult
         /// </remarks>
         [EditorBrowsable(EditorBrowsableState.Advanced)]
         public required ClaimsPrincipal? UserinfoTokenPrincipal { get; init; }
+
```
Comment (kevinchalet) [inline src/OpenIddict.Client/OpenIddictClientModels.cs]:
```suggestion
```

File: src/OpenIddict.Client/OpenIddictClientModels.cs [hunk 3]
```diff
@@ -253,6 +253,11 @@ public sealed record class ClientCredentialsAuthenticationResult
         /// </summary>
         public required string AccessToken { get; init; }
 
+        /// <summary>
+        /// Gets or sets the access token expiration date, if available.
```
Comment (kevinchalet) [inline src/OpenIddict.Client/OpenIddictClientModels.cs]:
```suggestion
        /// Gets or sets the expiration date of the access token, if available.
```

(for consistency with the comment used for `ProcessAuthenticationContext.BackchannelAccessTokenExpirationDate`)

**Review by kevinchalet — COMMENTED** [review]
Thanks for your PR.

There are other places where the expiration date could be set too. Could you please make sure every result that has a `BackchannelAccessToken`, `FrontchannelAccessToken` or `AccessToken` property also has an expiration date?

File: src/OpenIddict.Client/OpenIddictClientService.cs [hunk 4]
```diff
@@ -265,9 +265,11 @@ public async ValueTask<InteractiveAuthenticationResult> AuthenticateInteractivel
                     AuthorizationCode = context.AuthorizationCode,
                     AuthorizationResponse = context.Request is not null ? new(context.Request.GetParameters()) : new(),
                     BackchannelAccessToken = context.BackchannelAccessToken,
+                    BackchannelAccessTokenExpirationDate = context.BackchannelAccessTokenExpirationDate!,
```
Comment (kevinchalet) [inline src/OpenIddict.Client/OpenIddictClientService.cs]: Is there a particular reason for using the null forgiving operator? AFAICT, all the properties you added in this PR are nullable?

Comment (davhdavh) [reply src/OpenIddict.Client/OpenIddictClientService.cs]: None whatsoever, pushed fix

File: src/OpenIddict.Client/OpenIddictClientModels.cs [hunk 5]
```diff
@@ -66,6 +66,11 @@ public sealed record class InteractiveAuthenticationResult
         /// </summary>
         public required string? BackchannelAccessToken { get; init; }
 
+        /// <summary>
+        /// Gets or sets the expiration date of the back channel access token, if available.
```
Comment (kevinchalet) [inline src/OpenIddict.Client/OpenIddictClientModels.cs]:
```suggestion
        /// Gets or sets the expiration date of the backchannel access token, if available.
```

File: src/OpenIddict.Client/OpenIddictClientModels.cs [hunk 6]
```diff
@@ -82,6 +87,11 @@ public sealed record class InteractiveAuthenticationResult
         /// </summary>
         public required string? FrontchannelAccessToken { get; init; }
 
+        /// <summary>
+        /// Gets or sets the expiration date of the front channel access token, if available.
```
Comment (kevinchalet) [inline src/OpenIddict.Client/OpenIddictClientModels.cs]:
```suggestion
        /// Gets or sets the expiration date of the frontchannel access token, if available.
```

**Review by kevinchalet — APPROVED** [review]

**Comment by kevinchalet** [comment]
Merged. Congrats for your first contribution to OpenIddict 👍🏻

**Pull request merged**